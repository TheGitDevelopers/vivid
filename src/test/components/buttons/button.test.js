import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../../components/common/button/button';

describe('Button tests:', () => {
  /* Test mockups */
  const mockButtonName = 'button';

  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('click');

  const tree = renderer
    .create(<Button onPress={onPressEvent} text={mockButtonName} />);

  tree.root.findByType(Button).props.onPress();

  /* All component tests should be listed here */
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('has a text prop', () => {
    expect(tree.root.findByType(Button).props.text).toBe(mockButtonName);
  });
  it('has a pressable function', () => {
    expect(onPressEvent.mock.calls.length).toBe(1);
  });
  it('press function returns a value on click', () => {
    expect(onPressEvent.mock.results[0].value).toBe('click');
  });
});
