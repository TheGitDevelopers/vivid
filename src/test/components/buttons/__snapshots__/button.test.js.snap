// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests: renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#01E096",
          "borderRadius": 60,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 20,
          "width": "100%",
        },
        null,
        null,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "lato-regular",
          "fontSize": 16,
          "textTransform": "uppercase",
        }
      }
    >
      button
    </Text>
  </View>
</View>
`;
